src/Keboola/DbExtractor/Configuration/ConfigRowDefinition.php

```diff
@@ -79,9 +64,74 @@ public function getConfigTreeBuilder(): TreeBuilder
             ->end();
         // @formatter:on
 
+        // Defined Query and Table node at the same time
+        $rootNode->validate()
+            ->ifTrue(function ($v) {
+                return (isset($v['query']) && isset($v['table']));
+            })
+            ->thenInvalid('Both table and query cannot be set together.')
+            ->end();
+
+        // Undefined Query and Table node
+        $rootNode->validate()
+            ->ifTrue(function ($v) {
+                return (!isset($v['query']) && !isset($v['table']));
+            })
+            ->thenInvalid('One of table or query is required.')
+            ->end();
+
+        // Defined Table node contains required fields
+        $rootNode->validate()
+            ->ifTrue(function ($v) {
+                return (
+                    !isset($v['query'])
+                    && (!isset($v['table']['tableName'])
+                        || !isset($v['table']['schema'])
+                        || $v['table']['tableName'] === '')
+                );
+            })
+            ->thenInvalid('The table property requires "tableName" and "schema"')
+            ->end();
+
+        // Defined Incremental Fetching without Table node (advanced query)
+        $rootNode->validate()
```

Comment:
Nedává smysl, aby ten incremental fetching byl součástí tabulky, když bez ni nedává smysl? (schema, tableName, incrementalFetch)

Reply:
Tohle bychom mozna meli nadhodit do nejake diskuze nad config rows, zatim mi prijdou castecne zmatene a jestli si dobre pamatuju, tak sam k nim mas uz asi 3. pripominku.

Reply:
Připomínky obecně k config rows nebo k jejich implementaci tady? V každým případě mě obojí zajíma :-)

Reply:
Nechápu, jak to souvisí s ConfigRows.

Reply:
@tomasfejfar Ja to beru tak, ze novy config se ridi podle ConfigRows. Co to je nevym, zatim to beru jako mytus, kteremu davame zivot (kdyz to prezenu). Ale predpokladam, ze s temi ConfigRows prichazi nejaka struktura tech konfiguraci. To co ty tady zminujes je za me pripominka na to, abychom si tu strukturu vydefinovali lepe. Neni to otazka teto komponenty, ale i vsech ostatnich imho.

Reply:
@Actimel spousta novejch komponent na tom bez problémů frčí a celý to vzniklo tím, že spousta větších komponent má dost podobný featury (iterace položek v poli, disablování, pojmenování). Živý to je, zpětně kompatibilní to je 100% (kromě existujících UI), můžem si kdyžtak zavolat a ukážu, o co jde a jak to pasuje na stávající komponenty.

Reply:
@ondrejhlavacek Jasne to chapu proc to vzniklo. Spise mi jde o to, ze uz to nejakou spolecnou formu ma a kdyz tam budeme chtit nezo menit, tak budeme muset upravit uz existujici UI a potom ty definice configu.

Muzeme se spojit i s @tomasfejfar a trochu to probrat nebo si nechat objasnit.

Reply:
Definice configů se měnit nutně nemusí, se stávajcím to může být 1:1 kompatibilní.

Reply:
@tomasfejfar - (pokud spravne chapu o co ti jde) - v config rows dostane extractor v configu vzdycky jen jednu tabulku, timpadem je to 'jedno' (ale asi by bylo cistsi aby to bylo pod table, coz by znamenalo update mysql ex ui)

Reply:
@odinuv s klukama jsme si volali a z mýho pohledu je změna struktury konfigurace v tuhle chvíli kosmetická věc, která roztočí spirálu migrací, což bude veleomrd. pokud to není životně nutný, strukturu konfigurace bych neměnil.